### PR TITLE
Add read/write functions to xfeatures2d and normalize naming convention

### PR DESCRIPTION
**Pull request by augustinmanecy**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/xfeatures2d/src/surf.hpp [hunk 1]
```diff
@@ -23,6 +23,25 @@ class SURF_Impl : public SURF
                                int nOctaves = 4, int nOctaveLayers = 2,
                                bool extended = true, bool upright = false);
 
+    void read( const FileNode& fn)
+    {
+      fn["extended"] >> extended;
```
Comment (berak) [inline modules/xfeatures2d/src/surf.hpp:28]: what about `hessianThreshold` ?

**Comment by berak** [comment]
you'll have to [fix the java tests](https://pullrequest.opencv.org/buildbot/builders/precommit-contrib_linux64/builds/15360/steps/test_java/logs/junit-html#org.opencv.test.features2d) which currently expect either [empty](https://github.com/opencv/opencv/blob/master/modules/features2d/misc/java/test/SURFFeatureDetectorTest.java#L161) files, or a [different layout](https://github.com/opencv/opencv/blob/master/modules/features2d/misc/java/test/SURFFeatureDetectorTest.java#L146)

and there's a small general problem. 
loading an xml from a previous version (no params in it) will lead to resetting the algorithm's params to 0, example:

```
<?xml version="1.0"?>
<opencv_storage>
</opencv_storage>
```

```
FileStorage fs2("my.xml",0);
int nOctaves=4; 
fs2["nOctaves"] >> nOctaves;
// nOctaves is 0 now
```

**Comment by augustinmanecy** [comment]
@berak I think I corrected the java tests which failed in your link (TestCase Features2dTest.TestPTOD, and tests for BRIEF and SURF) 
However, as I never dealt with java before and I do not know how to run all of this tests before committing a validated fix ;p Can you give me some instructions to be able to execute the test on my side before committing again?

**Comment by alalek** [comment]
Build target: `opencv_test_java`
From build dir: `python <opencv_src_path>/modules/ts/misc/run.py -v -a -t java --package=xfeatures2d`
(package parameter is optional)

**Comment by augustinmanecy** [comment]
Hi, I just pushed a new commit which fix Java tests. I also added additionnal check in read methods in order to avoid erasing current value of parameters which are not in the configuration file (if a paramater was missing in the configuration file, it was replaced by 0). 
Normally tests should pass now.

**Pull request closed without merging**